packager/util/util.go

```diff
@@ -75,6 +76,17 @@ func hasCanSignHttpExchangesExtension(cert *x509.Certificate) bool {
 	return false
 }
 
+// Returns the Duration of time before cert expires with given deadline.
+// Returns -1 if cert is already expired. This will be used to periodically check if cert
+// is still within validity range.
+func GetDurationToExpiry(cert *x509.Certificate, currentTime time.Time) (time.Duration, error) {
+	if cert.NotBefore.After(currentTime) || cert.NotAfter.Before(currentTime) {
+		return time.Duration(-1), errors.New("Certificate has invalid duration")
```

Comment:
Go convention is to return the zero value on error. In this case, I think you can say `return 0, errors...`; the cast to `Duration` is implicit.